araig_interpreters/src/location_interpreters/vicon_interpreter.cpp

```diff
@@ -0,0 +1,135 @@
+#include <ros/ros.h>
+#include <tf/transform_broadcaster.h>
+#include <tf/transform_listener.h>
+#include <geometry_msgs/TransformStamped.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <std_msgs/String.h>
+#include <ros/master.h>
+#include <vector>
+#include <map>
+#include <string>
+#include <boost/algorithm/string.hpp>
+#include <boost/bind.hpp>
+
+class vicon_interpreter
+{
+private:
+    ros::NodeHandle nh;
+    ros::NodeHandle nh_priv;
+
+    std::vector<std::string> viconTopics;
+    ros::master::V_TopicInfo viconTopicInfos;
+
+    tf::TransformListener tf_ls;
+    tf::Transform trans_ls;
+
+    geometry_msgs::PoseStamped pose_msg;
+
+    std::map<std::string, ros::Publisher> viconPubList;
+    std::map<std::string, ros::Subscriber> viconSubList;
+
+    std::map<std::string, geometry_msgs::PoseStamped> OutputPoseMsg;
```

Comment:
consider refactoring variable name to concisely and precisely describe its functionality.